### ISSUE TITLE
Mutex Ctor Failing due to '\' in second argument

### ISSUE THREAD

**Issue by gleesonger** [alekdavis/SingleInstance#1]
Script is failing with the below error:
```
New-Object : Exception calling ".ctor" with "3" argument(s): "Could not find a part of the path 'C:\foo\bar.ps1'."
At \\somelocation\WindowsPowerShell\Modules\SingleInstance\1.1.0\SingleInstance.psm1:64 char:21
+ ... ipt:Mutex = New-Object System.Threading.Mutex($true, $ScriptId, [ref] ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [New-Object], MethodInvocationException
    + FullyQualifiedErrorId : ConstructorInvokedThrowException,Microsoft.PowerShell.Commands.NewObjectCommand
```

Using the filepath ($ScriptID) as the 'name' argument is causing the problem as a backslash is a special character in Mutex constructor, see this link for more info
[https://stackoverflow.com/questions/20714120/could-not-find-a-part-of-the-path-error-while-creating-mutex](https://stackoverflow.com/questions/20714120/could-not-find-a-part-of-the-path-error-while-creating-mutex)

**Comment by alekdavis** [comment]
@gleesonger: You are calling the cmdlet without specifying the script ID parameter, right? And your module is on a share identified by a UNC path (\\\\somelocation\\...), right? The problem is not the backslash. Backslashes work fine for mutexes. The problem is that since the caller does not pass the script ID, cmdlet tries to determine it based on the script path, and the script path is the path to the module itself, which includes two backslashes. First, of all, you don't want to do this. You do not want to set a mutex based on the module (since it can be called by different script). Second, just pass the ID from the caller (making sure it has no double back slashes). Not a bug in my mind, but when I have time I will update documentation and maybe do a safety check.

**Comment by gleesonger** [comment]
You are correct about my setup. A check with a useful error message would be helpful.
